### PR TITLE
PIP-2499: Update OpenTelemetry deps.

### PR DESCRIPTION
**Pull request by Baliedge**
https://mailgun.atlassian.net/browse/PIP-2499

* Bring OpenTelemetry deps up to date.
   * Required to resolve mixed version incompatibilities since updating to `go.opentelemetry.io/otel@v1.16.0`:
```
go: finding module for package go.opentelemetry.io/otel/metric/instrument
go: finding module for package go.opentelemetry.io/otel/metric/global
github.com/mailgun/blackbook/config imports
        github.com/mailgun/bunker/v11/client/golang imports
        go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc imports
        go.opentelemetry.io/otel/metric/global: module go.opentelemetry.io/otel/metric@latest found (v1.16.0), but does not contain package go.opentelemetry.io/otel/metric/global
github.com/mailgun/blackbook/config imports
        github.com/mailgun/bunker/v11/client/golang imports
        go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc imports
        go.opentelemetry.io/otel/metric/instrument: module go.opentelemetry.io/otel/metric@latest found (v1.16.0), but does not contain package go.opentelemetry.io/otel/metric/instrument
```

### CODE REVIEW

File: httpsign/random.go [hunk 1]
```diff
@@ -74,6 +74,7 @@ type SeededRNG struct {
 // generated by a math/rand.Rand.
 func (r *SeededRNG) bytes(bytes int) ([]byte, error) {
 	if r.rand == nil {
+		//nolint: gosec // TODO: use crypto/math random number generator.
```
Comment (vtopc) [inline httpsign/random.go:77]: There is the next comment a few lines above:

> Not cryptographically secure, not thread safe.

Comment (Baliedge) [reply httpsign/random.go]: Revised comment.

**Review by medge-sinch — APPROVED** [review]

File: consul/lock_test.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//nolint:gocritic // ignore sloppyTestFuncName
```
[the remark this answers is not in the document]
Comment (vtopc) [reply consul/lock_test.go:1]: Oh. Got it.

**Review by vtopc — APPROVED** [review]

File: consul/lock_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Baliedge) [reply consul/lock_test.go:1]: Here's what I get:
```
➜ ~/src/holster (Baliedge/PIP-2499-update-otel) ✗ rm $GOPATH/bin/golangci-lint
➜ ~/src/holster (Baliedge/PIP-2499-update-otel) ✗ make lint
curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /Users/spoulson/go/bin v1.52.2
golangci/golangci-lint info checking GitHub for tag 'v1.52.2'
golangci/golangci-lint info found version: 1.52.2 for v1.52.2/darwin/arm64
golangci/golangci-lint info installed /Users/spoulson/go/bin/golangci-lint
/Users/spoulson/go/bin/golangci-lint run
➜ ~/src/holster (Baliedge/PIP-2499-update-otel) ✗ vim consul/lock_test.go
➜ ~/src/holster (Baliedge/PIP-2499-update-otel) ✗ make lint
/Users/spoulson/go/bin/golangci-lint run
consul/lock_test.go:23:1: sloppyTestFuncName: function cleanUp should be of form TestXXX(t *testing.T) (gocritic)
func cleanUp(t *testing.T) {
^
make: *** [lint] Error 1
```

[the remark this answers is not in the document]
Comment (vtopc) [reply consul/lock_test.go:1]: e.g.
```
linters-settings:
  gocritic:
    # Which checks should be disabled; can't be combined with 'enabled-checks'.
    # Default: []
    disabled-checks:
      - sloppyTestFuncName
```

[the remark this answers is not in the document]
Comment (Baliedge) [reply consul/lock_test.go]: Agreed.  Added to `disabled-checks`.

**Pull request merged**